macros/models/_project/oneinch/oneinch_ar_calls_transfers_macro.sql

```diff
@@ -47,83 +50,102 @@ calls as (
     left join (
         select 
-            block_time
-            , tx_hash
-            , case
-                when value > uint256 '0' then 0xae
-                else "to"
-            end as token_address
-            , case {{selector}}
-                when {{transfer_selector}} then bytearray_to_uint256(substr(input, 37, 32)) -- transfer
-                when {{transfer_from_selector}} then bytearray_to_uint256(substr(input, 69, 32)) -- transferFrom
+            tx_hash
+            , if(value > uint256 '0', {{ wrapper_token_address }}, "to") as contract_address
+            , if(value > uint256 '0', true, false) as native_token
+            , case {{ selector }}
+                when {{ transfer_selector }} then bytearray_to_uint256(substr(input, 37, 32))
+                when {{ transfer_from_selector }} then bytearray_to_uint256(substr(input, 69, 32))
                 else value
             end as amount
             , case
-                when {{selector}} = {{transfer_selector}} or value > uint256 '0' then "from" -- transfer
-                when {{selector}} = {{transfer_from_selector}} then substr(input, 17, 20) -- transferFrom
+                when {{ selector }} = {{ transfer_selector }} or value > uint256 '0' then "from"
+                when {{ selector }} = {{ transfer_from_selector }} then substr(input, 17, 20)
             end as transfer_from
             , case
-                when {{selector}} = {{transfer_selector}} then substr(input, 17, 20) -- transfer
-                when {{selector}} = {{transfer_from_selector}} then substr(input, 49, 20) -- transferFrom
+                when {{ selector }} = {{ transfer_selector }} then substr(input, 17, 20)
+                when {{ selector }} = {{ transfer_from_selector }} then substr(input, 49, 20)
                 when value > uint256 '0' then "to"
             end as transfer_to
             , trace_address
             , input
             , success
             , call_type
-        from {{ source('bnb', 'traces') }}
+        from {{ source(blockchain, 'traces') }}
         where 
             {% if is_incremental() %}
-                block_time >= date_add('day', {{lookback_days}}, now())
+                block_time >= date_add('day', {{ lookback_days }}, now())
             {% else %}
                 block_time >= {{ project_start_date }}
             {% endif %}
-            and ({{selector}} in ({{transfer_selector}}, {{transfer_from_selector}}) or value > uint256 '0')
+            and ({{ selector }} in ({{ transfer_selector }}, {{ transfer_from_selector }}) or value > uint256 '0')
             and tx_success
             and success
     ) transfers on transfers.tx_hash = calls.tx_hash
-        and slice(transfers.trace_address, 1, cardinality(calls.start)) = calls.start
-        
+        and slice(transfers.trace_address, 1, cardinality(calls.start)) = calls.start 
 )
 
-select * from merged
+
+select 
+    *
+    , cast(tx_hash as varchar)||'--'||
```

Comment:
do you have a strong opinion on why you'd generate a surrogate key like this to use as unique key, rather than just using multiple columns and not introducing a new one?

Reply:
To keep `{blockchain}/oneinch_{blockchain}_ar_calls_transfers.sql` code and tests simpler. Looked at **nft_transfers** and did the same :)

Reply:
also, otherwise I'd need to create a "system" column like [here](https://github.com/grkhr/spellbook/blob/3a3fb96af389a402699125081d29caaa1f7c8750/models/oneinch/fantom/oneinch_fantom_calls_transfers.sql#L85) because it's nullable but be in primary key?

Reply:
makes sense -- thank you